src/components/Navigation.tsx

```diff
@@ -67,53 +168,45 @@ const SourceList: React.FC<SourceListProps> = ({ items }) => (
   </List>
 )
 
-type NavigationProps = {
-  items: [
-    string,
-    {
-      category?: string
-    }[]
-  ][]
-}
+export const Navigation: React.FC<NavigationProps> = ({ items }) => {
+  const ref = React.useRef()
+  const [menuOpen, setMenuOpen] = React.useState(false)
 
-export const Navigation: React.FC<NavigationProps> = ({ items }) => (
-  <Box
-    as="nav"
-    css={{
-      borderRight: '1px solid $tonal300',
-      boxSizing: 'border-box',
-      height: '100vh',
-      overflowX: 'hidden',
-      overflowY: 'auto',
-      p: '$3',
-      position: 'sticky',
-      top: 0,
-      width: 260
-    }}
-  >
-    <Heading size="sm" css={{ mb: '$4' }}>
-      Atom Learning
-      <br />
-      Design System
-    </Heading>
-    {Object.entries(transformNavigationStructure(items)).map(
-      ([source, content]) => (
-        <React.Fragment key={source}>
-          <SourceHeading>{capitalCase(source)}</SourceHeading>
-          {Array.isArray(content) ? (
-            <SourceList items={content} />
-          ) : (
-            Object.entries(content).map(([category, pages]) => (
-              <React.Fragment key={category}>
-                {category && category !== 'void' && (
-                  <CategoryHeading>{category}</CategoryHeading>
-                )}
-                <SourceList items={pages} />
-              </React.Fragment>
-            ))
-          )}
-        </React.Fragment>
-      )
-    )}
-  </Box>
-)
+  useOnClickOutside(ref, () => setMenuOpen(false))
+
+  return (
+    <>
+      <NavigationButton onClick={() => setMenuOpen(true)} />
+      <StyledNavigation ref={ref} open={menuOpen}>
+        <Heading size="sm" css={{ mb: '$4' }}>
+          Atom Learning
+          <br />
+          Design System
+        </Heading>
+        {Object.entries(items).map(([source, content]) => (
+          <React.Fragment key={source}>
+            <SourceHeading>{capitalCase(source)}</SourceHeading>
+            {Array.isArray(content) ? (
```

Comment:
This section of the markup is pretty noisy and I found the ternary difficult to parse. Could the section in the ternary be pulled into a separate component, which could then use two return statements to improve readability?